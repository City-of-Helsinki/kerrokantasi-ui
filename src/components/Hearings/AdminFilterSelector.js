import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../utils/Icon';
import {Nav, NavItem} from 'react-bootstrap';
import {FormattedMessage} from 'react-intl';

/**
 * [AdminFilterSelector description]
 * @param {object} props
 * @param {String} props.active
 * @param {function} props.onSelect
 * @param {Array<String|Object>} props.options
 */
const AdminFilterSelector = ({
  active,
  onSelect,
  options,
}) => {
  return (
    <Nav activeKey={active} bsStyle="pills" className="admin-filter-selector">
      {options.map((filter) => {
        const {list, iconName, formattedMessage, role} = filter;
        // NavItem role is button by default if filter.role doesnt exist
        return (
          <NavItem key={list} eventKey={list} onSelect={onSelect} role={role}>
            <Icon name={iconName} aria-hidden />
            <FormattedMessage id={formattedMessage}>{txt => txt}</FormattedMessage>
          </NavItem>
        );
      })}
    </Nav>
  );
};

AdminFilterSelector.propTypes = {
  active: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.shape({
      list: PropTypes.string,
      formattedMessage: PropTypes.string,
    }))
  ]),
};

export default AdminFilterSelector;
