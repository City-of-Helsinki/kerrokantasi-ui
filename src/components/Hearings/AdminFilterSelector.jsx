import React from 'react';
import PropTypes from 'prop-types';
import { SelectionGroup, RadioButton } from 'hds-react';
import { FormattedMessage, useIntl } from 'react-intl';

/**
 * [AdminFilterSelector description]
 * @param {object} props
 * @param {String} props.active
 * @param {function} props.onSelect
 * @param {Array<String|Object>} props.options
 */
const AdminFilterSelector = ({ active, onSelect, options }) => {
  const intl = useIntl();

  return (
    <SelectionGroup
      direction='horizontal'
      className='admin-filter-selector'
      label={intl.formatMessage({ id: 'adminFilterSelectorLabel' })}
    >
      {options.map((filter) => {
        const { list, formattedMessage } = filter;
        return (
          <RadioButton
            key={list}
            id={`filter-${list}`}
            value={list}
            checked={active === list}
            onChange={() => onSelect(list)}
            label={<FormattedMessage id={formattedMessage}>{(txt) => txt}</FormattedMessage>}
          />
        );
      })}
    </SelectionGroup>
  );
};

AdminFilterSelector.propTypes = {
  active: PropTypes.string,
  onSelect: PropTypes.func,
  options: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(
      PropTypes.shape({
        list: PropTypes.string,
        formattedMessage: PropTypes.string,
      }),
    ),
  ]),
};

export default AdminFilterSelector;
