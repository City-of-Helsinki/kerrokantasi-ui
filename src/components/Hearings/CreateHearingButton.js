import React from 'react';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';
import {Button, ButtonToolbar} from 'react-bootstrap';
import Icon from '../../utils/Icon';
import Link from '../LinkWithLang';

const CreateHearingButton = ({to}) => (
  <div className="toolbar-bottom create-hearing-button-container">
    <ButtonToolbar className="actions pull-right">
      <Link to={to}>
        <Button bsStyle="success">
          <Icon name="plus"/>&nbsp;&nbsp;
          <FormattedMessage id="createHearing"/>
        </Button>
      </Link>
    </ButtonToolbar>
  </div>
);

CreateHearingButton.propTypes = {
  to: PropTypes.object.isRequired
};

export default CreateHearingButton;
