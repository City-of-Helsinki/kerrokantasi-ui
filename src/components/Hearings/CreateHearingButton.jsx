import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Button } from 'hds-react';

import Icon from '../../utils/Icon';
import Link from '../LinkWithLang';

const CreateHearingButton = ({ to }) => (
  <div className='toolbar-bottom create-hearing-button-container'>
    <div className='actions pull-right'>
      <Link to={to}>
        <Button className='kerrokantasi-btn success'>
          <Icon name='plus' />
          &nbsp;&nbsp;
          <FormattedMessage id='createHearing' />
        </Button>
      </Link>
    </div>
  </div>
);

CreateHearingButton.propTypes = {
  to: PropTypes.object.isRequired,
};

export default CreateHearingButton;
