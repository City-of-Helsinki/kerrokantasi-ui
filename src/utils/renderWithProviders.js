/* eslint react/prop-types: 0 */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';
import { IntlProvider } from 'react-intl';

import createStore from "../createStore";
import { userOidcConfig, apiTokenClientConfig } from './oidcConfig';

const loginProviderConfig = {
  userManagerSettings: userOidcConfig,
  apiTokensClientSettings: apiTokenClientConfig,
  sessionPollerSettings: { pollIntervalInMs: 10000 },
}

const renderWithProviders = (
  ui,
  { preloadedState = {}, locale = 'fi', history, store = createStore(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider {...loginProviderConfig}>
      <IntlProvider locale={locale}>
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </LoginProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};


export default renderWithProviders;
