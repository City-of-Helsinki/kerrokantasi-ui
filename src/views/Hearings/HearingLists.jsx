const HearingLists = {
  ALL: {
    list: 'allHearings',
    isFetching: false,
    params: {},
    formattedMessage: 'allHearings',
    iconName: 'globe',
  },
  OPEN: {
    list: 'openHearings',
    isFetching: false,
    params: {},
    formattedMessage: 'openHearings',
    iconName: 'commenting-o',
  },
  PUBLISHED: {
    list: 'publishedHearings',
    isFetching: false,
    params: { published: 'True' },
    formattedMessage: 'publishedHearings',
    iconName: 'eye',
  },
  QUEUE: {
    list: 'publishingQueueHearings',
    isFetching: false,
    params: { published: 'True' },
    formattedMessage: 'publishingQueue',
    iconName: 'calendar-check-o',
  },
  DRAFTS: {
    list: 'draftHearings',
    isFetching: false,
    params: { published: 'False' },
    formattedMessage: 'drafts',
    iconName: 'pencil-square-o',
  },
  OWN: {
    list: 'ownHearings',
    isFetching: false,
    params: {},
    formattedMessage: 'ownHearings',
    iconName: 'user',
    role: 'link',
  },
};

export default HearingLists;

export const AdminFilters = [HearingLists.PUBLISHED, HearingLists.QUEUE, HearingLists.DRAFTS, HearingLists.OWN];