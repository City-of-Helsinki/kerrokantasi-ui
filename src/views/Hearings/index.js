// TODO: remove this disable once https://github.com/yannickcr/eslint-plugin-react/pull/1628 lands
/* eslint-disable react/no-unused-prop-types */
/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import Helmet from 'react-helmet';
import { Col, Row } from 'react-bootstrap';
import { get, find, includes } from 'lodash';
import isEmpty from 'lodash/isEmpty';
import { withRouter } from 'react-router-dom';
import * as Actions from '../../actions';
import { isAdmin } from '../../utils/user';
import WrappedHearingList from '../../components/HearingList';
import LoadSpinner from '../../components/LoadSpinner';
import CreateHearingButton from '../../components/Hearings/CreateHearingButton';
import AdminFilterSelector from '../../components/Hearings/AdminFilterSelector';
import { hearingShape, labelShape, userShape } from '../../types';
import getAttr from '../../utils/getAttr';
import { parseQuery, stringifyQuery } from '../../utils/urlQuery';
import {getUser} from '../../selectors/user';

const now = () => new Date().toISOString();

const HearingLists = {
  ALL: {
    list: 'allHearings',
    params: {},
    formattedMessage: 'allHearings',
    iconName: 'globe'
  },
  OPEN: {
    list: 'openHearings',
    params: {},
    formattedMessage: 'openHearings',
    iconName: 'commenting-o'
  },
  PUBLISHED: {
    list: 'publishedHearings',
    params: { published: 'True' },
    formattedMessage: 'publishedHearings',
    iconName: 'eye'
  },
  QUEUE: {
    list: 'publishingQueueHearings',
    params: { published: 'True' },
    formattedMessage: 'publishingQueue',
    iconName: 'calendar-check-o'
  },
  DRAFTS: {
    list: 'draftHearings',
    params: { published: 'False' },
    formattedMessage: 'drafts',
    iconName: 'pencil-square-o'
  },
  OWN: {
    list: 'ownHearings',
    params: {},
    formattedMessage: 'ownHearings',
    iconName: 'user',
    role: 'link'
  }
};

const AdminFilters = [HearingLists.PUBLISHED, HearingLists.QUEUE, HearingLists.DRAFTS, HearingLists.OWN];

const getHearingListParams = listName => {
  const defaultParams = { include: 'geojson' };
  const hearingList = find(HearingLists, ({ list }) => list === listName);
  const hearingListParams = get(hearingList, 'params', {});

  const params = Object.assign({}, defaultParams, hearingListParams);

  switch (listName) {
    case HearingLists.PUBLISHED.list:
      params.open_at_lte = now();
      break;
    case HearingLists.QUEUE.list:
      params.open_at_gt = now();
      break;
    default:
    // Don't edit params further
  }
  return params;
};

export class Hearings extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      sortBy: '-created_at',
      adminFilter: isAdmin(props.user) ? AdminFilters[0].list : null,
      showOnlyOpen: false,
    };

    this.handleSearch = this.handleSearch.bind(this);
    this.handleSelectLabels = this.handleSelectLabels.bind(this);
    this.handleSort = this.handleSort.bind(this);
    this.setAdminFilter = this.setAdminFilter.bind(this);
    this.toggleShowOnlyOpen = this.toggleShowOnlyOpen.bind(this);
  }

  componentDidMount() {
    const { fetchLabels } = this.props;
    fetchLabels();
    // Hearing List is fetched when labels are available -> componentWillReceiveProps
  }

  componentWillReceiveProps(nextProps) {
    const { user, location, match: {params: {tab}}, labels } = this.props;
    const { adminFilter } = this.state;
    const shouldSetAdminFilter = isAdmin(nextProps.user) && (!user || !adminFilter);
    const shouldNullAdminFilter = isAdmin(user) && !nextProps.user;
    const shouldFetchHearings = labels && ((
      (!this.props.labels.length && nextProps.labels.length) ||
      (nextProps.labels.length && location.search !== nextProps.location.search) ||
      (!this.props.user && nextProps.user) ||
      (this.props.user && !nextProps.user)) || nextProps.match.params.tab !== tab);

    if (shouldSetAdminFilter) {
      this.setAdminFilter(AdminFilters[0].list);
    }

    if (shouldNullAdminFilter) {
      this.setAdminFilter(null);
    }

    if (shouldFetchHearings) {
      this.fetchHearingList(nextProps);
    }
  }

  getHearings() {
    const { hearingLists } = this.props;
    const hearingListKey = this.getHearingListName();
    return get(hearingLists, [hearingListKey, 'data'], null);
  }

  getHearingsCount() {
    const { hearingLists } = this.props;
    const list = this.getHearingListName();
    return get(hearingLists, [list, 'count'], 0);
  }

  getHearingListName() {
    const { user } = this.props;
    const { adminFilter } = this.state;

    return isAdmin(user) ? adminFilter : HearingLists.ALL.list;
  }

  getIsLoading() {
    const { hearingLists } = this.props;
    const name = this.getHearingListName();
    return get(hearingLists, [name, 'isFetching'], true);
  }

  static getSearchParams(props) {
    const { location } = props;
    const params = {};

    if (parseQuery(location.search).search) Object.assign(params, {title: parseQuery(location.search).search});
    if (parseQuery(location.search).label) {
      Object.assign(params, {label: Hearings.getLabelsFromQuery(parseQuery(location.search).label.toString())});
    }
    return params;
  }

  setAdminFilter(filter) {
    if (filter === 'ownHearings') {
      this.forwardToUserHearings();
    } else {
      this.setState({ adminFilter: filter }, () => this.fetchHearingList());
    }
  }

  forwardToUserHearings() {
    const {history, location} = this.props;
    const searchParams = parseQuery(location.search);
    history.push({
      pathname: '/user-hearings',
      search: stringifyQuery(searchParams)
    });
  }

  fetchHearingList(props = this.props) {
    const {fetchInitialHearingList, fetchAllHearings, match: {params: {tab}}} = props;
    const { sortBy } = this.state;
    const list = this.getHearingListName();
    const params = Object.assign({}, getHearingListParams(list), { ordering: sortBy }, Hearings.getSearchParams(props));

    if (tab === 'map') {
      fetchAllHearings(list, params);
    } else {
      fetchInitialHearingList(list, params);
    }
  }

  static getLabelsFromQuery = (labelsInQuery = []) => {
    if (Array.isArray(labelsInQuery)) return labelsInQuery;

    return [labelsInQuery];
  };

  handleSearch(searchTitle, force = false) {
    const { history, location } = this.props;
    const searchParams = parseQuery(location.search);
    const searchParamsEmpty = isEmpty(searchParams.search);
    const searchPhraseEmpty = isEmpty(searchTitle);
    const searchPhraseUpdated = searchParams.search !== searchTitle;

    // Don't do anything if:
    // search is not forced
    // AND
    // current search params are the same as the inputted search
    // OR
    // both search params and currently inputted search are empty
    if (!force && ((searchParams.search === searchTitle) || (searchParamsEmpty && searchPhraseEmpty))) {
      return;
    }

    if (searchPhraseEmpty) {
      delete searchParams.search;
    } else {
      searchParams.search = searchTitle;
    }

    if (searchPhraseUpdated || force) {
      history.push({
        path: location.path,
        search: stringifyQuery(searchParams),
      });
    }
  }

  handleSelectLabels(labels) {
    const { history, location } = this.props;
    const searchParams = parseQuery(location.search);
    searchParams.label = labels.map(({ id }) => id);
    history.push({
      path: location.pathname,
      search: stringifyQuery(searchParams)
    });
  }

  handleSort(sortBy) {
    this.setState(
      () => ({ sortBy }),
      () => {
        this.fetchHearingList();
      },
    );
  }

  toggleShowOnlyOpen() {
    this.setState(({ showOnlyOpen }) => ({ showOnlyOpen: !showOnlyOpen }));
  }

  handleReachBottom = () => {
    const {fetchMoreHearings, hearingLists} = this.props;
    const list = this.getHearingListName();

    if (
      hearingLists[list] &&
      hearingLists[list].count > hearingLists[list].data.length &&
      typeof hearingLists[list].next === 'string' &&
      !hearingLists[list].isLoading
    ) {
      fetchMoreHearings(list);
    }
  }

  render() {
    const {
      history,
      intl: { formatMessage },
      labels,
      language,
      match: { params: { tab } },
      location,
      user,
    } = this.props;
    const labelsInQuery = Array.isArray(parseQuery(location.search).label)
      ? parseQuery(location.search).label
      : [parseQuery(location.search).label];
    const selectedLabels = labels
      .filter(label => includes(labelsInQuery, label.id.toString())).map(label => getAttr(label.label, language));
    const searchTitle = parseQuery(location.search).search;
    const { showOnlyOpen } = this.state;
    const hearings = this.getHearings();
    const hearingCount = this.getHearingsCount();

    if (user && user.isFetching) {
      return <LoadSpinner />;
    }
    return (
      <div className="hearings">
        <section className="page-section page-section--all-hearings-header">
          <div className="container">
            <Row>
              <Col md={10} mdPush={1}>
                <Helmet title={formatMessage({ id: 'allHearings' })} />
                <FormattedMessage id="allHearings">
                  {allHearings => (
                    <h1
                      className="page-title"
                      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
                      tabIndex="0"
                      aria-label={allHearings}
                      id="allHearingsPageTitle"
                    >
                      {allHearings}
                    </h1>
                  )}
                </FormattedMessage>
                {isAdmin(user) &&
                  <AdminFilterSelector
                  onSelect={this.setAdminFilter}
                  options={AdminFilters}
                  active={this.getHearingListName()}
                  />
                }
                {isAdmin(user) && <CreateHearingButton to={{path: '/hearing/new'}} />}
              </Col>
            </Row>
          </div>
        </section>
        {!isEmpty(labels) ? <WrappedHearingList
          hearings={hearings}
          hearingCount={hearingCount}
          selectedLabels={selectedLabels ? [].concat(selectedLabels) : []}
          searchPhrase={searchTitle}
          isLoading={this.getIsLoading()}
          labels={labels}
          showOnlyOpen={showOnlyOpen}
          handleSort={this.handleSort}
          handleSearch={this.handleSearch}
          handleSelectLabels={this.handleSelectLabels}
          toggleShowOnlyOpen={this.toggleShowOnlyOpen}
          language={language}
          tab={tab}
          intl={this.props.intl}
          handleReachBottom={this.handleReachBottom}
          onTabChange={value => {
            const url = `/hearings/${value}`;
            history.push({
              pathname: url,
              search: location.search,
            });
          }}
        /> : <LoadSpinner />}
      </div>
    );
  }
}

Hearings.propTypes = {
  fetchInitialHearingList: PropTypes.func,
  fetchMoreHearings: PropTypes.func,
  fetchLabels: PropTypes.func,
  hearingLists: PropTypes.objectOf(
    PropTypes.shape({
      isFetching: PropTypes.boolean,
      data: PropTypes.arrayOf(hearingShape),
    }),
  ),
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
  intl: intlShape.isRequired,
  labels: PropTypes.arrayOf(labelShape),
  language: PropTypes.string,
  location: PropTypes.shape({
    search: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  }),
  match: PropTypes.shape({
    params: PropTypes.shape({
      tab: PropTypes.string,
    }),
  }),
  user: PropTypes.shape({
    data: userShape,
    isFetching: PropTypes.boolean,
  }),
};

const mapStateToProps = state => ({
  hearingLists: state.hearingLists,
  labels: state.labels.data,
  language: state.language,
  user: getUser(state), // Using this on purpose without selector
});

const mapDispatchToProps = dispatch => ({
  fetchInitialHearingList: (listName, params) =>
    dispatch(Actions.fetchInitialHearingList(listName, '/v1/hearing/', params)),
  fetchAllHearings: (list, params) => dispatch(Actions.fetchHearingList(list, '/v1/hearing', params)),
  fetchMoreHearings: (listName) => dispatch(Actions.fetchMoreHearings(listName)),
  fetchLabels: () => dispatch(Actions.fetchLabels()),
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(injectIntl(Hearings)));
