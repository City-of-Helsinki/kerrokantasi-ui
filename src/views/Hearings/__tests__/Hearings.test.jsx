import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import Hearings from '../index';
import { mockStore, getIntlAsProp, mockUser } from '../../../../test-utils';
import renderWithProviders from '../../../utils/renderWithProviders';
import createAppStore from '../../../createStore';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/hearings/list',
    search: '',
  }),
}));

const renderComponent = (propOverrides) => {
  const { labels, hearingLists, ...rest } = mockStore;

  const props = {
    labels: labels.data,
    ...rest,
    ...propOverrides,
  };

  return renderWithProviders(
    <MemoryRouter>
      <Hearings {...props} />
    </MemoryRouter>
  );
};

describe('<Hearings />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render HearingList when labels are present', async () => {
    renderComponent();

    expect(await screen.findAllByRole('listitem')).toHaveLength(1);
  });
});
