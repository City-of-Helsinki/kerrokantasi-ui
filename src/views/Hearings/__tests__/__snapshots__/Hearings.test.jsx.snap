// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hearings /> should render correctly 1`] = `
<div>
  <div
    class="hearings"
  >
    <section
      class="page-section page-section--all-hearings-header"
    >
      <div
        class="all-hearings-header-container"
      >
        <h1
          class="page-title"
        >
          allHearings
        </h1>
      </div>
    </section>
    <section
      class="hearings-list"
    >
      <section
        class="page-section--hearings-search"
      >
        <div
          class="hearings-search__container"
        >
          <div
            class="hearings-search__content"
          >
            <form
              id="hearings-search-form"
            >
              <div
                class="hearings-search__controls"
              >
                <div
                  class="hearings-search__text form-group"
                >
                  <div
                    class="SearchInput-module_root__sMwrY hearings-search__input"
                  >
                    <label
                      class="FieldLabel-module_label__1zrXK "
                      for="downshift-0-input"
                      id="downshift-1-label"
                    >
                      <span>
                        searchTitles
                      </span>
                    </label>
                    <div
                      class="SearchInput-module_wrapper__114JK"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="downshift-1-menu"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-labelledby="downshift-1-label"
                        aria-owns="downshift-1-menu"
                        autocomplete="off"
                        class="SearchInput-module_input__Addns"
                        enterkeyhint="search"
                        id="downshift-0-input"
                        role="combobox"
                        value=""
                      />
                      <div
                        class="SearchInput-module_buttons__2O_2t search-input_hds-search-input__buttons__3PawT"
                      >
                        <button
                          aria-label="Etsi otsikoista"
                          class="SearchInput-module_button__z7NIp search-input_hds-search-input__button__1-cj-"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            aria-label="search"
                            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik SearchInput-module_searchIcon__l1_S4"
                            role="img"
                            viewBox="0 0 24 24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M15 1C19.4183 1 23 4.58172 23 9C23 13.4183 19.4183 17 15 17C13.2001 17 11.539 16.4056 10.2023 15.4023L3.80101 21.8021L2.21002 20.2111L8.60822 13.8117C7.59861 12.4726 7.00002 10.8062 7.00002 9C7.00002 4.58172 10.5817 1 15 1ZM15 3C11.6863 3 9.00002 5.68629 9.00002 9C9.00002 12.3137 11.6863 15 15 15C18.3137 15 21 12.3137 21 9C21 5.68629 18.3137 3 15 3Z"
                              fill="currentColor"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </button>
                      </div>
                      <ul
                        aria-labelledby="downshift-1-label"
                        class="SearchInput-module_menu__2P8CO"
                        id="downshift-1-menu"
                        role="listbox"
                        style="max-height: 416px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="hearings-search__label form-group"
                >
                  <label
                    class="control-label"
                    for="formControlsSearchSelect"
                  >
                    <span>
                      searchLabels
                    </span>
                  </label>
                  <div
                    class="Select hearings-search__select is-clearable is-searchable Select--multi"
                  >
                    <div
                      class="Select-control"
                    >
                      <div
                        class="Select-multi-value-wrapper"
                        id="react-select-2--value"
                      >
                        <div
                          class="Select-placeholder"
                        >
                          Etsi...
                        </div>
                        <div
                          class="Select-input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-activedescendant="react-select-2--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-owns=""
                            id="formControlsSearchSelect"
                            role="combobox"
                            style="box-sizing: content-box; width: 5px;"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: 100%; font-family: inherit; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                      <span
                        class="Select-arrow-zone"
                      >
                        <span
                          class="Select-arrow"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO hearings-search__button kerrokantasi-btn"
                type="submit"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <span>
                    search
                  </span>
                </span>
              </button>
              <a
                class="internal-link hidden-link"
                href="#hearings-section"
              >
                <span>
                  jumpToSearchResults
                </span>
              </a>
            </form>
          </div>
        </div>
      </section>
      <section
        class="hearing-list-tabs"
      >
        <div
          class="Tabs-module_tabs__2Ww5F custom-theme-1"
        >
          <div
            class="Tabs-module_tablistBar__elMv9 page-section--hearings-tabs"
          >
            <div
              class="Tabs-module_tablist__1iuDO"
            >
              <ul
                role="tablist"
                style="width: 0px;"
              >
                <li
                  aria-controls="tab-0-panel"
                  aria-selected="true"
                  class="Tabs-module_tab__3oOoX Tabs-module_active__1Ic3B"
                  id="tab-0-button"
                  role="tab"
                  tabindex="0"
                >
                  <span>
                    <span>
                      list
                    </span>
                  </span>
                </li>
                <li
                  aria-controls="tab-1-panel"
                  aria-selected="false"
                  class="Tabs-module_tab__3oOoX"
                  id="tab-1-button"
                  role="tab"
                  tabindex="-1"
                >
                  <span>
                    <span>
                      map
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            aria-labelledby="tab-0-button"
            class="hearings-list-tab-panel"
            id="tab-0-panel"
            role="tabpanel"
          >
            <section
              class="hearings-list-tab-panel-container"
            >
              <a
                class="internal-link hidden-link"
                href="#hearings-search-form"
              >
                <span>
                  jumpToSearchForm
                </span>
              </a>
              <div>
                <div
                  class="hearing-list__result-controls"
                >
                  <h2
                    class="hearing-list__hearing-list-title"
                  >
                    totalNumHearings
                  </h2>
                  <div
                    class="hearing-list__filter-bar clearfix"
                  >
                    <div
                      class="sr-only"
                    />
                    <div
                      class="hearing-list__filter-bar-filter"
                      id="formControlsSelect"
                    >
                      <div
                        class="Select-module_root__Ka5uO"
                      >
                        <label
                          class="FieldLabel-module_label__1zrXK "
                          for="hds-select-2-toggle-button"
                          id="hds-select-2-label"
                        >
                          <span>
                            sort
                          </span>
                        </label>
                        <div
                          class="Select-module_wrapper__1WQXs"
                        >
                          <button
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-labelledby="hds-select-2-label hds-select-2-toggle-button"
                            aria-owns="hds-select-2-menu"
                            class="Select-module_button__1aIsm"
                            id="hds-select-2-toggle-button"
                            type="button"
                          >
                            <span
                              class="Select-module_buttonLabel__1fqu5"
                            >
                              Uusimmat ensin
                            </span>
                            <svg
                              aria-hidden="true"
                              aria-label="angle-down"
                              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
                              role="img"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                                fill="currentColor"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </button>
                          <ul
                            aria-labelledby="hds-select-2-label"
                            class="Select-module_menu__1H2aU"
                            id="hds-select-2-menu"
                            role="listbox"
                            style="max-height: 260px;"
                            tabindex="-1"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <ul
                  class="hearing-list"
                >
                  <li
                    class="hearing-list-item"
                  >
                    <div
                      alt="Mock Hearing"
                      class="mouse-only-click-element hearing-list-item-image"
                      style="background-image: url(https://api.hel.fi/kerrokantasi-test/media/images/2017/10/LzqVMrMI.jpeg);"
                    />
                    <div
                      class="hearing-list-item-content"
                    >
                      <div
                        class="hearing-list-item-title-wrap"
                      >
                        <h2
                          class="h4 hearing-list-item-title"
                        >
                          <a
                            href="/mockHearing?headless=false"
                          >
                             
                            Mock Hearing
                          </a>
                        </h2>
                        <div
                          class="hearing-list-item-comments"
                        >
                          <span
                            aria-hidden="true"
                            class="fa fa-comment-o"
                          />
                           
                          1
                          <span
                            class="sr-only"
                          >
                            <span>
                              hearingListComment
                            </span>
                          </span>
                        </div>
                      </div>
                      <div
                        class="hearing-list-item-times"
                      >
                        <div>
                          <span>
                            timeOpenPastWithValues
                          </span>
                        </div>
                        <div>
                          <span>
                            timeCloseFutureWithValues
                          </span>
                        </div>
                      </div>
                      <div
                        class="hearing-list-item-labels clearfix"
                      >
                        <div
                          aria-label="labelListTagsTitle"
                          class="hearing-list-item-labellist"
                        />
                      </div>
                      <div
                        class="hearing-card-notice"
                      >
                        <span
                          aria-hidden="true"
                          class="fa fa-exclamation-circle"
                        />
                        <span>
                          hearingTranslationNotAvailable
                        </span>
                        <div
                          class="language-available-message"
                          lang="fi"
                        >
                          Kuuleminen saatavilla suomeksi
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
                <div
                  class="loader-wrap"
                  data-testid="load-spinner"
                >
                  <span
                    class="fa fa-hourglass-o fa-2x fa-spin"
                  />
                </div>
              </div>
            </section>
          </div>
        </div>
      </section>
    </section>
  </div>
</div>
`;
